ant-networking/src/event/swarm.rs

```diff
@@ -506,6 +515,15 @@ impl SwarmDriver {
                     }
                 };
 
+                // Just track failures during outgoing connection with `failed_peer_id` inside the bootstrap cache.
+                // OutgoingConnectionError without peer_id can happen when dialing multiple addresses of a peer.
+                // And similarly IncomingConnectionError can happen when a peer has multiple transports/listen addrs.
+                if let (Some((_, failed_addr, _)), Some(bootstrap_cache)) =
```

Comment:
I think this block of code can be moved into the below block of `if should_clean_peer {` ?
where a peer is for sure hits a serious issue and got removed from the RT ?